Implemented Data Editor Svelte Test Structure

stricklandrbls opened the pull request:
- Added yarn script to run data editor tests using Mocha.
- Added example function and Svelte store tests.

This is a simple directory structure and test suite setup that utilizes the `mocha` test suite that's currently a utilized dependency. The `test:svelte` script should be ran manually during development and is not currently implemented to be ran during packaging of the extension.

With the `mocha -w` option, it does not currently support _traditional_ sourcing of the directory due a discrepancy with with node. [^1] Typing `r s <Enter>`, while the `test:svelte` script is watching, will manually source the directory structure and re-run the tests. [^2]

[^1]: https://mochajs.org/#nodejs-native-esm-support
[^2]: https://mochajs.org/#-watch-w

Closes #915

shanedell commented on package.json:
```diff
@@ -41,6 +41,7 @@
     "package": "run-func build/yarn-scripts.ts package && yarn --cwd dist/package install && yarn --cwd dist/package vsce package --out ../../",
     "pretest": "yarn compile && yarn webpack",
     "test": "sbt test && node ./out/tests/runTest.js",
```
I would say that either before or after `node ./out/tests/runTest.js` you want to run the svelte tests. So either,
```json
"test": "sbt test && yarn test:svelte && node ./out/tests/runTest.js",
```
or
```json
"test": "sbt test && node ./out/tests/runTest.js && yarn test:svelte",
```
I personally prefer the first option but either way is fine with me also. Once that is updated I think this is good to go.

stricklandrbls replied:
I initially did not have the tests being ran during the projects tests because there were a lot of compilation errors within the svelte tests. This is because of #878 

Although, I found out that there are only compilation errors when `mocha` is trying to compile the files with the errors. The example tests that I have do not have any so I went ahead and implemented your changes to the test scripts.

stricklandrbls commented:
@shanedell I added a new script & modified 2 existing scripts:
- Added `watch:svelte-tests` - For having only the svelte tests watched during development.
- Changed `test:svelte` to not have the `-w` option for watching.
- Changed `test` to your first suggestion.

stricklandrbls commented:
> @shanedell I added a new script & modified 2 existing scripts:
> 
>     * Added `watch:svelte-tests` - For having only the svelte tests watched during development.
> 
>     * Changed `test:svelte` to not have the `-w` option for watching.
> 
>     * Changed `test` to your first suggestion.

Looks like the `mocha` side is still failing on the CI side due to the same compilation errors I mentioned before. I'm not entirely sure why though because running `mocha -r ts-node/register ./src/svelte/tests/**/*.test.ts` or `yarn test:svelte` does not throw those errors on my machine.

nlewis05 commented:
My local build is passing
![image](https://github.com/apache/daffodil-vscode/assets/131286323/8cfcd3c1-35b0-4547-a6ee-88190fe30496)

shanedell reviewed (approved):
+1
